utils/HelperFunctions.ts

```diff
@@ -22,54 +13,61 @@ function clearObject(obj) {
 //Adds a random offset to a number within a certain percentage
 //e.g. addOffset(100, 5) will return anything from 95 to 105.
 //The percentage argument must be between 0 and 100;
-function addOffset(n, percentage) {
+function addOffset(n: number, percentage: number): number {
     if (percentage < 0 || percentage > 100) {return n;}
 
-    var offset = n * (percentage / 100);
+    const offset = n * (percentage / 100);
 
     return n + ((Math.random() * (2 * offset)) - offset);
 }
 
 //Given an element by its Id(usually an 'a' element), removes all event listeners
 //from that element by cloning and replacing. Then returns the new cloned element
-function clearEventListeners(elemId) {
-    var elem = document.getElementById(elemId);
-    if (elem == null) {console.log("ERR: Could not find element for: " + elemId); return null;}
-    var newElem = elem.cloneNode(true);
-    elem.parentNode.replaceChild(newElem, elem);
-    return newElem;
+function clearEventListeners(elemId: string): HTMLElement | null {
+    const elem: HTMLElement | null = document.getElementById(elemId);
+    if(elem === null) return null;
+    return clearEventListenersEl(elem as HTMLElement);
 }
 
 //Same as clearEventListeners except it takes a DOM element object rather than an ID
-function clearEventListenersEl(el) {
+function clearEventListenersEl(el: HTMLElement): HTMLElement | null {
     if (el == null) {console.log("ERR: element passed into clearEventListenersEl is null"); return null;}
-    var newElem = el.cloneNode(true);
-    el.parentNode.replaceChild(newElem, el);
-    return newElem;
+    const newElem: Node = el.cloneNode(true);
+    if(el.parentNode === null) return null;
+    (el.parentNode as Element).replaceChild(newElem, el);
+    return newElem as HTMLElement;
 }
 
 //Given its id, this function removes an element AND its children
-function removeElementById(id) {
-    var elem = document.getElementById(id);
-    if (elem == null) {return;}
-    while(elem.firstChild) {elem.removeChild(elem.firstChild);}
-    elem.parentNode.removeChild(elem);
+function removeElementById(id: string) {
+    const elem: HTMLElement | null = document.getElementById(id);
+    if(elem === null) return;
+    removeElement(elem as HTMLElement);
 }
 
-function removeElement(elem) {
-    if (elem == null || !(elem instanceof Element)) {return;}
+// Same as removeElementById except if takes a DOM element object rather than an ID
+function removeElement(elem: Element) {
```

Comment:
why the use of "Element" vs. "HTMLElement"? Usage seems to be mixed. I'm sure there is a hierarchy between them, just not sure which is the more general and if we really need it to be.